evm/contracts/PrimarySaleClaimdropFactory.sol

```diff
@@ -0,0 +1,207 @@
+    /// @notice Initialize factory with owner
+    /// @param initialOwner Address of the initial owner
+    /// @param name_ Human-readable name of the factory
+    /// @param slug_ URL-friendly identifier
+    /// @param description_ Description of the factory's purpose
+    function initialize(
+        address initialOwner,
+        string memory name_,
+        string memory slug_,
+        string memory description_
+    ) public initializer {
+        __Ownable_init(initialOwner);
+        __Pausable_init();
+        
+        name = name_;
+        slug = slug_;
+        description = description_;
+    }
+
+    // ============ External Functions ============
+
+    /// @notice Deploy a new Claimdrop contract
+    /// @dev Owner of the new Claimdrop will be the factory owner
+    /// @dev Requires PrimarySale to be set before deploying Claimdrops
+    /// @return claimdropAddress Address of the newly deployed Claimdrop
+    function deployClaimdrop() external onlyOwner whenNotPaused returns (address claimdropAddress) {
+        if (primarySale == address(0)) revert PrimarySaleNotSet();
+        
+        // Deploy new Claimdrop with factory owner as the owner
+        Claimdrop claimdrop = new Claimdrop(owner());
+        claimdropAddress = address(claimdrop);
+
+        // Track the deployed contract
+        deployedClaimdrops.push(claimdropAddress);
+        isClaimdrop[claimdropAddress] = true;
+
+        emit ClaimdropDeployed(primarySale, claimdropAddress, owner(), deployedClaimdrops.length - 1);
+    }
+
+    /// @notice Set the PrimarySale contract address (only one allowed)
+    /// @dev PrimarySale must be deployed externally to avoid contract size limits
+    /// @param primarySale_ Address of the deployed PrimarySale contract
+    function setPrimarySale(address primarySale_) external onlyOwner whenNotPaused {
+        if (primarySale != address(0)) revert PrimarySaleAlreadyDeployed();
+        if (primarySale_ == address(0)) revert InvalidAddress();
+
+        primarySale = primarySale_;
+
+        emit PrimarySaleSet(primarySale_, owner());
+    }
+
+    /// @notice Get the total number of deployed Claimdrops
+    /// @return count Total number of deployed Claimdrops
+    function getDeployedClaimdropsCount() external view returns (uint256 count) {
+        return deployedClaimdrops.length;
+    }
+
+    /// @notice Get all deployed Claimdrop addresses
+    /// @return addresses Array of all deployed Claimdrop addresses
+    function getAllDeployedClaimdrops() external view returns (address[] memory addresses) {
+        return deployedClaimdrops;
+    }
+
+    /// @notice Get a specific deployed Claimdrop by index
+    /// @param index Index in the deployedClaimdrops array
+    /// @return claimdropAddress Address of the Claimdrop at the given index
+    function getClaimdropAtIndex(uint256 index) external view returns (address claimdropAddress) {
+        require(index < deployedClaimdrops.length, "Index out of bounds");
```

Comment:
Inconsistent Error Handling in PrimarySaleClaimdropFactory.sol

  - Issue: getClaimdropAtIndex() uses old-style require() with string message instead of custom errors
  - Code:
  require(index < deployedClaimdrops.length, "Index out of bounds");
  - Expected Pattern: All other errors in the repo use custom errors (lines 101, 118-119)
  - Impact: Inconsistent error handling reduces code quality and increases bytecode size
  - Recommendation: Replace with:
  error IndexOutOfBounds();
  // Then in function:
  if (index >= deployedClaimdrops.length) revert IndexOutOfBounds();